### PR TITLE
Improve gitversion.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: tools/gitversion.cmake [hunk 1]
```diff
@@ -98,7 +101,7 @@ function(compute_git_version VERSION)
       set(${VERSION} "${DEFAULT_PRE_VERSION}-pre.${COMMIT_COUNT}" PARENT_SCOPE)
       return()
     endif()
-    set(RELEASE_BRANCHES "/release-${LATEST_VERSION_TAG}")
+  #  set(RELEASE_BRANCHES "/release-${LATEST_VERSION_TAG}")
```
Comment (kasperl) [inline tools/gitversion.cmake:104]: In or out?

File: tools/gitversion.cmake [hunk 2]
```diff
@@ -226,13 +232,48 @@ function(compute_git_version VERSION)
...
+        # Add our commit count at the end.
+        # For example, if the latest tag is v2.0.0-alpha.39, then we generate v2.0.0-alpha.39.<commit-count>.
+        set(RESULT_PREFIX "${LATEST_VERSION_TAG}.${COMMITS_SINCE_LAST_TAG}")
+      else()
+        # A normal version number.
+        # Add a '-pre.<commit-count>' suffix.
+        set(RESULT_PREFIX "${LATEST_VERSION_TAG}-pre.${COMMITS_SINCE_LAST_TAG}")
+      endif()
+    else()
+      message("not greater")
```
Comment (kasperl) [inline tools/gitversion.cmake:264]: Is this useful?

File: tools/gitversion.cmake [hunk 3]
```diff
@@ -226,13 +232,48 @@ function(compute_git_version VERSION)
...
+    set(branch_minor 0)
+  else()
+    list(GET RELEASE_BRANCHES 0 NEWEST_BRANCH)
+    version_from_branch("${NEWEST_BRANCH}" branch_major branch_minor)
+  endif()
+
+  if (NOT "${LATEST_VERSION_TAG}" STREQUAL "")
+    string(SUBSTRING "${LATEST_VERSION_TAG}" 1 -1 LATEST_VERSION_TAG_NO_V)
+    if (LATEST_VERSION_TAG_NO_V VERSION_GREATER "${branch_major}.${branch_minor}.0")
+          # Count the commits since the tag.
```
Comment (kasperl) [inline tools/gitversion.cmake:249]: Weird indentation.

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: tools/gitversion.cmake (hunk 1 above)
Comment (floitsch) [reply tools/gitversion.cmake]: done.

File: tools/gitversion.cmake (hunk 3 above)
Comment (floitsch) [reply tools/gitversion.cmake]: done.

File: tools/gitversion.cmake (hunk 2 above)
Comment (floitsch) [reply tools/gitversion.cmake]: done.

**Pull request merged**